code/__HELPERS/unsorted.dm

```diff
@@ -1517,3 +1517,20 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	for(var/V in GLOB.player_list)
 		if(is_servant_of_ratvar(V) || isobserver(V))
 			. += V
+
+//src may be null, but it does need to be a typed var
+#define NAMEOF(datum, X) (list(datum.X, #X)[2])
+
+#define VARSET_LIST_CALLBACK(target, var_name, var_value) CALLBACK(GLOBAL_PROC, /proc/___timervarset, target, var_name, var_value)
+//dupe code because dm can't handle deep macros
```

Comment:
'deep macros'?

Reply:
if you use macro A, which uses macro B, which uses macro C the compiler gives up

Reply:
honestly don't believe you, I think you've just fudged the logic since the preprocessor is VERY specific about what it'll accept (needs braces, multiple lines need escaping, semicolons, etc.)